### PR TITLE
rose edit: allow newline separators in space-separated list

### LINKED ISSUE #1660: rose edit: allow newline separators in space-separated list
A space-separated list variable in `rose edit` can display oddly if the separators are newlines.

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #1660 

@matthewrmshin Please review.

### CODE REVIEW

File: lib/python/rose/config_editor/valuewidget/array/spaced_list.py [hunk 1]
```diff
@@ -424,7 +424,7 @@ def spaced_array_join(values):
 def spaced_array_split(value):
     """Split the value into elements with appropriate string values."""
     try:
-        value_array = value.split(" ")
+        value_array = re.split(r'[\s\n]+', value)
```
Comment (matthewrmshin) [inline lib/python/rose/config_editor/valuewidget/array/spaced_list.py]: Should use `shlex.split` here to handle quoted strings.

Comment (oliver-sanders) [inline lib/python/rose/config_editor/valuewidget/array/spaced_list.py]: Done

**Comment by matthewrmshin** [comment]
@arjclark please sanity check.

**Comment by arjclark** [comment]
Seems ok to me

**Pull request merged**